File: program/find-gcd-of-two-numbers-by-recursion/FindGcdOfTwoNumbersByRecursion.java
import java.io.*;
import java.util.Scanner;

class GCD {
    static int gcd(int a, int b) {
        if (b == 0)
            return a;
        return gcd(b, a % b);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int a = sc.nextInt();
        int b = sc.nextInt();
        sc.close();

        System.out.println("GCD of " + a + " and " + b + " is " + gcd(a, b));
    }
}

File: program/find-gcd-of-two-numbers-by-recursion/FindGcdOfTwoNumbersByRecursion.cs
using System;

class GCD {
  static int gcd(int x, int y) {
    if (y == 0)
      return x;
    return gcd(y, x%y);
  } 

  static void Main(string[] args) {
    int x = 12;
    int y = 18;

    Console.WriteLine("GCD of "+ x +" and "+ y +" is: "+ gcd(x,y));
  }
}
